internal/serialization/serialization.go

```diff
@@ -201,16 +242,19 @@ func (s *Service) registerSerializer(serializer pubserialization.Serializer) err
 	return nil
 }
 
-func (s *Service) registerClassDefinitions(portableSerializer *PortableSerializer,
-	classDefinitions []*pubserialization.ClassDefinition) {
-	for _, cd := range classDefinitions {
-		portableSerializer.portableContext.RegisterClassDefinition(cd)
+func (s *Service) registerClassDefinitions(serializer *PortableSerializer, classDefs []*pubserialization.ClassDefinition) {
+	for _, cd := range classDefs {
+		if err := serializer.portableContext.RegisterClassDefinition(cd); err != nil {
+			panic(err)
+		}
 	}
 }
 
 func (s *Service) registerGlobalSerializer(globalSerializer pubserialization.Serializer) {
 	if globalSerializer != nil {
-		s.registerSerializer(globalSerializer)
+		if err := s.registerSerializer(globalSerializer); err != nil {
+			panic(err)
+		}
```

Comment:
Not related to this, but I wasn't able to put comments there. We don't need `getIDByObject` anymore

Reply:
Removed at: https://github.com/hazelcast/hazelcast-go-client/pull/581/commits/646f1893fc76711d3ffecc2c6d09d0253044d316